test/Parse/diagnostic_call_at_type_level.swift

```diff
@@ -0,0 +1,12 @@
+// RUN: %target-typecheck-verify-swift
+
+struct Bar { // expected-note {{in declaration of 'Bar'}}
+ var fisr = 0x5F3759DF
+ var a: Int
+
+ // ensure that the id.id pattern is not interpreted as a function
+ a.foo = 345 // expected-error {{expected declaration}}
+ // ensure that the id.id pattern generating an expected declaration
+ // diagnostic does not block further diagnostics.
+ fisr.bar = 345
+}
```

Comment:
1. We can move this test into the `diagnostic_missing_func_keyword` file, since it's very much related.
2. You could give the struct a more descriptive name like `SR13098`.